flask_pymongo/__init__.py

```diff
@@ -183,12 +186,21 @@ def get_upload(filename):
         response.headers["Content-Disposition"] = f"attachment; filename={filename}"
         response.content_length = fileobj.length
         response.last_modified = fileobj.upload_date
-        # Compute the sha1 sum of the file for the etag.
-        pos = fileobj.tell()
-        raw_data = fileobj.read()
-        fileobj.seek(pos)
-        digest = hashlib.sha1(raw_data).hexdigest()
-        response.set_etag(digest)
+
+        # GridFS does not manage its own checksum, so we manage our own using its
+        # metadata storage, to be used for the etag.
+        sha1_sum = self._hash_cache.get(str(fileobj._id))
```

Comment:
This is problematic because although `_id` is unique, `_id`s can be reused over time by deleting the old file and uploading a new one with the same `_id`. In that case this cache becomes  out of date.